extensions/annotations/src/main/java/io/opentelemetry/extension/annotations/Timed.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This annotation creates a {@link io.opentelemetry.api.metrics.LongHistogram Histogram} instrument
+ * observing the duration of invocations of the annotated method or constructor.
+ *
+ * <p>By default, the Histogram instrument will have the following attributes:
+ *
+ * <ul>
+ *   <li><b>class:</b> The fully qualified name of the class whose method is invoked.
+ *   <li><b>method:</b> The name of the annotated method, or "new" of the annotation is on a
+ *       constructor.
+ *   <li><b>exception:</b> The {@link Class#getSimpleName() simple name} of the Exception if an
+ *       Exception is thrown, or "None" if the method did not throw an Exception.
```

Comment:
Hmm, can we use `exception.type` and `Class#getCanonicalName()` instead? It's what `Span#recordException()` uses.

Reply:
Thanks for the heads-up. If that's the standard attribute names from the Spec we should use them.

Reply:
I pushed an update using `exception.type` and the canonical name.